src/options.ts

```diff
@@ -139,10 +134,57 @@ export function _setDefaultOptions(options: Partial<Options> = {}): Options {
   };
 }
 
-export function defaultSpanExporterFactory(options: Options): JaegerExporter {
+export function resolveTracesExporter(): SpanExporterFactory {
+  const factory =
+    SpanExporterMap[process.env.OTEL_TRACES_EXPORTER ?? 'default'];
+  assert.strictEqual(
+    typeof factory,
+    'function',
+    `Invalid value for OTEL_TRACES_EXPORTER env variable: ${util.inspect(
+      process.env.OTEL_TRACES_EXPORTER
+    )}. Pick one of ${util.inspect(Object.keys(SpanExporterMap), {
+      compact: true,
+    })} or leave undefined.`
+  );
```

Comment:
nit: IMO the following would be much simpler to read for most people including me but up to you if you want to change

```javascript

const factory = spanExporterMap[process.env.OTEL_TRACES_EXPRTER ?? 'default']
if  (factory == undefined) {
    throw new Error(...)
}
return factory
```

Reply:
If you refer to the usage of `assert` then `if (...) { throw new Error(...); }` is exactly what that does - I do agree that it's somewhat less common outside of tests than if-error, but I do prefer that over if-errors if that's fine by you.
Or do you mean that the error message is long-winding?

Reply:
I don't mind the assert per se but I feel simple `if` condition to be much easier read and understand than a function that takes callbacks, type of object, a literal "function", etc. Anyone not an expert in JS/TS will have to spent a few minutes trying to understand it and probably read docs on assert function. On the other hand, if condition is universal and needs 5 second to understand even if you've never written JS before.